### PR TITLE
update permission for tidb-controller-manager and add example for tidb-monitor

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 

1. After upgrade TiDB Operator from v1.0 to v1.1, then create TiDB Monitor CR, tidb-controller-manager reports below error:
```
Event(v1.ObjectReference{Kind:"TidbMonitor", Namespace:"dan10", Name:"dan10", UID:"c8b9b9ed-67f9-11ea-8886-8a678ff421f7", APIVersion:"pingcap.com/v1alpha1", ResourceVersion:"68470646", FieldPath:""}): type: 'Warning' reason: 'FailedSync' Sync TidbMonitor[dan10/dan10] Deployment failed,err:secrets "dan10-monitor" is forbidden: User "system:serviceaccount:tidb-admin:tidb-controller-manager" cannot update resource "secrets" in API group "" in the namespace "dan10"
 ```
2. No full example for TiDB Monitor.

### What is changed and how does it work?
update permission for tidb-controller-manager and add example for tidb-monitor
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
   TiDB Monitor CR can be created and synced.

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

Other issues during test of TiDB Monitor are recorded in #1949.

### CODE REVIEW

File: charts/tidb-operator/templates/controller-manager-rbac.yaml [hunk 1]
```diff
@@ -38,7 +38,7 @@ rules:
   verbs: ["get", "list", "watch", "create", "update", "delete"]
 - apiGroups: [""]
   resources: ["secrets"]
-  verbs: ["create", "get", "list", "watch"]
+  verbs: ["create", "update", "get", "list", "watch"]
```
Comment (weekface) [inline charts/tidb-operator/templates/controller-manager-rbac.yaml]: does this should be consistent with https://github.com/pingcap/tidb-operator/pull/1954/files#diff-ecdc286a1f57077063e4a16696198f7fR138?

File: manifests/monitor/tidb-monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
+apiVersion: pingcap.com/v1alpha1
```
Comment (weekface) [inline manifests/monitor/tidb-monitor.yaml]: these is already an example of TidbMonitor: `examples/basic/tidb-monitor.yaml`

Comment (DanielZhangQD) [reply manifests/monitor/tidb-monitor.yaml]: Those are basic examples and users cannot know how to use the CR with the basic ones.

File: charts/tidb-operator/templates/controller-manager-rbac.yaml (hunk 1 above)
Comment (DanielZhangQD) [reply charts/tidb-operator/templates/controller-manager-rbac.yaml]: updated.

**Comment by DanielZhangQD** [comment]
> I recommend to omit empty fields, given that we have a detailed api reference now.

It would be easier for users to know how to use the CR following this example without referring to the API doc.
Comments addressed, PTAL again, thanks! @weekface @aylei

**Pull request merged**